fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordKeyCursorTest.java

```diff
@@ -0,0 +1,270 @@
+            // Note that the PK start with 1, so the location is one-off when removed
+            store.deleteRecord(result.get(16).getPrimaryKey());
+            store.deleteRecord(result.get(21).getPrimaryKey());
+            store.deleteRecord(result.get(22).getPrimaryKey());
+            store.deleteRecord(result.get(70).getPrimaryKey());
+            commit(context);
+        }
+        // Scan records
+        ScanProperties scanProperties = getScanProperties(useContinuations);
+        final List<Tuple> actualKeys;
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            actualKeys = scanKeys(store, scanProperties, useContinuations != UseContinuations.NONE);
+            commit(context);
+        }
+        List<Tuple> expectedKeys = IntStream.range(1, 101).filter(i -> !Set.of(17, 22, 23, 71).contains(i)).boxed().map(Tuple::from).collect(Collectors.toList());
+        assertEquals(expectedKeys, actualKeys);
+    }
+
+    public static Stream<Arguments> splitNumberContinuationsVersion() {
+        return Stream.of(0, 1, 2, 3)
+                .flatMap(
+                        splitNumber -> Stream.of(UseContinuations.NONE, UseContinuations.CONTINUATIONS, UseContinuations.BYTE_LIMIT)
+                                .flatMap(useContinuastions -> Stream.of(3, 6, FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION)
+                                        .map(formatVersion -> Arguments.of(splitNumber, useContinuastions, formatVersion))));
+    }
+
+    @ParameterizedTest(name = "testIterateRecordsMissingSplit [splitNumber = {0}, useContinuations = {1}, formatVersion = {2}]")
+    @MethodSource("splitNumberContinuationsVersion")
+    public void testIterateRecordsMissingSplit(int splitNumber, UseContinuations useContinuations, int formatVersion) throws Exception {
+        boolean splitLongRecords = true;
+
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> savedRecords = saveRecords(splitLongRecords, formatVersion, hook);
+        // Delete a split
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            // If operating on the short record, #0 is the only split
+            // If operating on the long record, splits can be 1,2,3
+            // Use splitNumber to decide which record to operate on.
+            // Record #1 in the saved records is a short record, #16 is a long (split) record
+            int recordNumber = (splitNumber == 0) ? 1 : 16;
+            final Tuple pkAndSplit = savedRecords.get(recordNumber).getPrimaryKey().add(splitNumber);
+            byte[] split = store.recordsSubspace().pack(pkAndSplit);
+            store.ensureContextActive().clear(split);
+            commit(context);
+        }
```

Comment:
To validate the test is doing what we expect, perhaps it would be worth trying to read the record we corrupted, and making sure we can't.

Reply:
I'm not sure I follow: If we fail the validation, it means that this record would show up in the listing of invalid records and would attempt a "repair". Why would failure to read be adding coverage?

Reply:
Done - read the records and ensure they fail.